resteasy-client-vertx/src/test/java/org/jboss/resteasy/test/client/vertx/VertxClientEngineTest.java

```diff
@@ -0,0 +1,348 @@
+
+      final CompletionStage<String> cs = client().target(baseUri()).request()
+         .header("Password", "abracadabra").rx(CompletionStageRxInvoker.class)
+         .get(String.class);
+
+      String response = cs.toCompletableFuture().get();
+      assertEquals("Success", response);
+   }
+
+   @Test
+   public void testBigly() throws Exception {
+      server.requestHandler(new EchoHandler());
+      final byte[] valuableData = randomAlpha().getBytes(StandardCharsets.UTF_8);
+      final Response response = client().target(baseUri()).request()
+            .post(Entity.entity(valuableData, MediaType.APPLICATION_OCTET_STREAM_TYPE));
+
+      assertEquals(200, response.getStatus());
+      assertArrayEquals(valuableData, response.readEntity(byte[].class));
+   }
+
+   @Test
+   public void testFutureResponse() throws Exception {
+      server.requestHandler(new EchoHandler());
+      final String valuableData = randomAlpha();
+      final Future<Response> response = client().target(baseUri()).request()
+            .buildPost(Entity.entity(valuableData, MediaType.APPLICATION_OCTET_STREAM_TYPE))
+            .submit();
+
+      final Response resp = response.get(10, TimeUnit.SECONDS);
+      assertEquals(200, resp.getStatus());
+      assertEquals(valuableData, resp.readEntity(String.class));
+   }
+
+   @Test
+   public void testFutureString() throws Exception {
+      server.requestHandler(new EchoHandler());
+      final String valuableData = randomAlpha();
+      final Future<String> response = client().target(baseUri()).request()
+            .buildPost(Entity.entity(valuableData, MediaType.APPLICATION_OCTET_STREAM_TYPE))
+            .submit(String.class);
+
+      final String result = response.get(10, TimeUnit.SECONDS);
+      assertEquals(valuableData, result);
+   }
+
+   private String randomAlpha() {
+      final StringBuilder builder = new StringBuilder();
+      final Random r = new Random();
+      for (int i = 0; i < 20 * 1024 * 1024; i++) {
+         builder.append((char) ('a' + (char) r.nextInt('z' - 'a')));
+         if (i % 100 == 0) builder.append('\n');
+      }
+      return builder.toString();
+   }
+
+   @Test
+   public void testCallbackString() throws Exception {
+      server.requestHandler(new EchoHandler());
+      final String valuableData = randomAlpha();
+      CompletableFuture<String> cf = new CompletableFuture<>();
+      client().target(baseUri()).request()
+          .buildPost(Entity.entity(valuableData, MediaType.APPLICATION_OCTET_STREAM_TYPE))
+          .submit(new InvocationCallback<String>() {
+             @Override
+             public void completed(String s) {
+                cf.complete(s);
+             }
+             @Override
+             public void failed(Throwable throwable) {
+                cf.completeExceptionally(throwable);
+             }
+          });
+
+      final String result = cf.get(10, TimeUnit.SECONDS);
+      assertEquals(valuableData, result);
+   }
+/*
+   @Test
```

Comment:
Should this one be removed?

Reply:
I forgot to resolve this, will try to make it work

Reply:
done and pushed